pkg/api/service.go

```diff
@@ -91,18 +164,23 @@ func (s *Service) SubscribeEnvelopes(
 	}
 
 	ch := s.subscribeWorker.listen(stream.Context(), query)
+	err = s.catchUpFromCursor(stream, query, log)
```

Comment:
I spent a lot of time thinking about whether there is a race condition here that would cause a 'gap' in payloads between when the catch-up ends and when the next batch of envelopes comes in from the subscribe worker. I *think* there is no race condition, here is my reasoning:

```
SUBSCRIBE WORKER ALGORITHM
- In a loop:
   - Fetch new payloads from DB
   - Fetch listeners
   - Push to listener channels
   - Wait 100ms
```

```
SUBSCRIBE REQUEST ALGORITHM (this function)
- Add self to subscribe worker listeners list
- Catch up from DB
- Pull from the listener channel
```

In order for there to be a race condition, the subscribe worker needs to fetch from the DB *after* the subscribe request's catch-up, but *not* push to the subscribe request's listener. I think that's not possible, because the subscribe request's listener was added to the subscribe worker's listener list before the catch-up was started.